database/sql/src/main/scala/cromwell/database/slick/MetadataSlickDatabase.scala

```diff
@@ -502,4 +502,19 @@ class MetadataSlickDatabase(originalDatabaseConfig: Config)
     val action = dataAccess.metadataArchiveStatusByWorkflowId(workflowId).result.headOption
     runTransaction(action).map(_.flatten)
   }
+
+  override def queryWorkflowsToArchiveThatEndedOnOrBeforeThresholdTimestamp(workflowStatuses: List[Option[String]],
```

Comment:
Does it make sense for this function to replace `queryWorkflowIdsByArchiveStatusAndEndedOnOrBeforeThresholdTimestamp` above? The only difference seems to be one vs. many statuses and call sites for the older function could simply be updated to pass a one-element `List(myStatus)`

Reply:
Hmmm, they are kind of similar but I think different enough to have separate methods. The `queryWorkflowIdsByArchiveStatusAndEndedOnOrBeforeThresholdTimestamp` is looking for workflows by metadata archive status (mostly Archived) while `queryWorkflowsToArchiveThatEndedOnOrBeforeThresholdTimestamp` is looking for workflows that are unarchived and have terminal workflow state. Also this method is looking for the oldest workflow to archive (using summary journal id) while `queryWorkflowIdsByArchiveStatusAndEndedOnOrBeforeThresholdTimestamp` isn't.

Reply:
Does writing this new method make the old one permanently unused though? I thought I remembered at one point Adam's PR said "this is unused but about to be, so let's not remove it yet"

Reply:
That was a different area, I think it was something to do with computing whether something is a root workflow

Reply:
The other method Adam was mentioning about, `queryWorkflowIdsByArchiveStatusAndEndedOnOrBeforeThresholdTimestamp`, is used by `DeleteMetadataActor` to look for next workflow to delete